react/components/ProductSummaryBuyButton.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import classNames from 'classnames'
+import BuyButton from 'vtex.store-components/BuyButton'
+import { equals, path } from 'ramda'
+import { FormattedMessage } from 'react-intl'
+
+import displayButtonTypes from '../DisplayButtonTypes'
+import productSummary from '../productSummary.css'
+
+const ProductSummaryBuyButton = ({
+  product,
+  displayMode,
+  displayBuyButton,
+  isOneClickBuy,
+  buyButtonText,
+  runtime: { hints: { mobile } },
+  isHovering,
+}) => {
+  const buyButtonClasses = classNames(
+    `${productSummary.buyButtonContainer} pv3 w-100`,
+    {
+      'dn': displayMode === 'small' || displayMode === 'inline',
+      'dn db-ns': displayMode === 'normal',
+    }
+  )
+
+  const showBuyButton = !equals(displayBuyButton, displayButtonTypes.DISPLAY_ON_HOVER.value) || mobile || isHovering
+  const quantity = path(['sku', 'seller', 'commertialOffer', 'AvailableQuantity'], product) || 0
+  const isAvailable = (quantity > 0)
+
+  return (
+    !equals(displayBuyButton, displayButtonTypes.DISPLAY_NONE.value) && (
```

Comment:
move this to an if statement before the return, react will complain if this component returns `false`. you should return `null` instead.